### ISSUE TITLE
Cannot automatically import multiple levels of nested directories under `utils`

### ISSUE THREAD

**Issue by noootwo** [nitrojs/nitro#2918]
### Environment

- Operating System: Linux
- Node Version:     v18.20.3
- Nitro Version:    2.10.4
- Package Manager:  npm@10.2.3

### Reproduction

https://stackblitz.com/edit/github-pxnqrn?file=server%2Futils%2Findex.ts,server%2Futils%2Ffoo%2Fbar%2Findex.ts,server%2Futils%2Ffoo%2Findex.ts,server%2Froutes%2Findex.ts

### Describe the bug

Cannot automatically import multi-level of nested directories under `server/utils`.

In `unimport`version `3.13` and earlier, auto-importing worked for two levels. 
However, in `unimport` version `3.14`, it only works for one level.

### Additional context

Is automatically import of multi-level(>3) directories expected behavior?

**Comment by noootwo** [comment]
@pi0 If auto import multi-level directories is the desired behavior, I can submit a PR to fix this issue

**Comment by antfu** [comment]
> /cc @antfu any clues? (i haven't checked anything yet but imagined you might have better context for resolution 🙏🏼 )

Explaining the context:
@noootwo worked on the PRs to `unimport` that changed a bit of how `dirs`'s root is resolved, which landed in `unimport@3.14` and caused some behavior change in nitro. It was because Nitro put `utils/*` in `dirs` (https://github.com/nitrojs/nitro/blob/575144f0efa9e8115251abb8d0bd09165ecd55f2/src/core/config/resolvers/imports.ts#L30) instead of `utils`. Previously, it was resolved as `utils/**/*.{ts,js}`, but now it is resolved to `utils/*.{ts,js}`. This issue was mainly asking about what's your original expectation of it, if you was expecting `utils` to resolve to nested folder, we would fix it in unimport.

My two cents:
I think we should fix that in `unimport` to align with the previous behavior. Otherwise it would be a breaking change to existing Nitro users anyway. Even if it was not a designed behavior (which to me it seems to expect to be deep), we could figure out a way to smooth it out and make the breakage in a major of nitro.

**Comment by noootwo** [comment]
> > /cc @antfu any clues? (i haven't checked anything yet but imagined you might have better context for resolution 🙏🏼 )
> 
> ##### Explaining the context:
> @noootwo worked on the PRs to `unimport` that changed a bit of how `dirs`'s root is resolved, which landed in `unimport@3.14` and caused some behavior change in nitro. It was because Nitro put `utils/*` in `dirs` (
> 
> https://github.com/nitrojs/nitro/blob/575144f0efa9e8115251abb8d0bd09165ecd55f2/src/core/config/resolvers/imports.ts#L30
> 
> ) instead of `utils`. Previously, it was resolved as `utils/**/*.{ts,js}`, but now it is resolved to `utils/*.{ts,js}`. This issue was mainly asking about what's your original expectation of it, if you was expecting `utils` to resolve to nested folder, we would fix it in unimport.
> ##### My two cents:
> I think we should fix that in `unimport` to align with the previous behavior. Otherwise it would be a breaking change to existing Nitro users anyway. Even if it was not a designed behavior (which to me it seems to expect to be deep), we could figure out a way to smooth it out and make the breakage in a major of nitro.

Correct a small error described here, previously, it was resolved as `utils/*` and `utils/*/*.{ts,js}`, so it can match the one and two levels files of the `utils` directory.

So if expect to match deep-level files, there were also some issues before.: )

**Comment by noootwo** [comment]
So I tend to prefer maintaining the current `unimport` behavior,  that better aligns with the intuitive understanding of  the `dirs` option. 

This means that need to change `utils/*` to `utils/**/*` in `nitro`.

If my understanding is off track, please correct me, thanks.

**Comment by antfu** [comment]
Yeah I agree that only two-level nesting is confusing, it should be either one level or deep. /cc @pi0 WDYT?

**Comment by pi0** [comment]
fixed on nightly channel and next release.
